packages/teams-js/test/public/visualMedia.spec.ts

```diff
@@ -0,0 +1,517 @@
+import { errorLibraryNotInitialized, permissionsAPIsRequiredVersion } from '../../src/internal/constants';
+import { GlobalVars } from '../../src/internal/globalVars';
+import { DOMMessageEvent } from '../../src/internal/interfaces';
+import { visualMedia } from '../../src/public';
+import { app } from '../../src/public/app';
+import { errorNotSupportedOnPlatform, FrameContexts } from '../../src/public/constants';
+import { DevicePermission, ErrorCode } from '../../src/public/interfaces';
+import { setUnitializedRuntime } from '../../src/public/runtime';
+import { Utils } from '../utils';
+
+/* eslint-disable */
+/* As part of enabling eslint on test files, we need to disable eslint checking on the specific files with
+   large numbers of errors. Then, over time, we can fix the errors and reenable eslint on a per file basis. */
+/**
+ * Test cases for visualMedia APIs
+ */
+describe('visualMedia', () => {
+  const originalDefaultPlatformVersion = '1.6.0';
+  const allowedContexts = [FrameContexts.content, FrameContexts.task];
+  const minVersionForPermissionsAPIs = permissionsAPIsRequiredVersion;
+  const errorInvalidCount = 'Invalid input count: Must supply the valid image counts (limit to 10).';
+  const cameraProps: visualMedia.CameraProps = {
+    source: visualMedia.Source.Camera,
+  };
+  const galleryProps: visualMedia.GalleryProps = {
+    source: visualMedia.Source.Gallery,
+  };
+  const commonImageprops: visualMedia.image.ImageProperties = {
+    shouldAllowInkingOnImages: false,
+    shouldAllowTextStickersOnImages: false,
+    enableFilterOnImages: false,
+    imageFormatsAllowedToBeSelected: [visualMedia.image.ImageOutputFormats.Image],
+  };
+  const defaultCaptureImageInputs: visualMedia.image.CameraImageProperties = {
+    maxVisualMediaCount: 1,
+    sourceProps: cameraProps,
+    commonImageprops: commonImageprops,
+  };
+  const defaultUploadImageInputs: visualMedia.image.GalleryImageProperties = {
+    maxVisualMediaCount: 1,
+    sourceProps: galleryProps,
+    commonImageprops: commonImageprops,
+  };
+  const defaultImageFiles: visualMedia.VisualMediaFile = {
+    id: 'uuid',
+    content: 'fake_content',
+    sizeInKB: 1,
+    name: 'fileName',
+    mimeType: 'jpg',
+  };
+  let utils: Utils = new Utils();
+  beforeEach(() => {
+    utils = new Utils();
+    utils.mockWindow.parent = undefined;
+    utils.messages = [];
+    GlobalVars.isFramelessWindow = false;
+  });
+  afterEach(() => {
+    app._uninitialize();
+    jest.clearAllMocks();
+    GlobalVars.isFramelessWindow = false;
+  });
+
+  describe('Testing isSupported', () => {
```

Comment:
This code is repeated from line 266 to 269. Please remove the redundant code.

Reply:
Thanks, that's reasonable! Removed.